jobs/job_utils.py

```diff
@@ -51,13 +57,31 @@ def clone_model(model, **new_values):
         Subclasses of BaseModel will return a clone with the same type.
     """
     # Reference implementation: https://stackoverflow.com/a/2712401/4859885.
+    model_id = new_values.pop('id', None) or get_model_id(model)
     cls = model.__class__
-    model_id = new_values.pop('id', model.id)
     props = {k: v.__get__(model, cls) for k, v in cls._properties.items()} # pylint: disable=protected-access
     props.update(new_values)
     return cls(id=model_id, **props)
 
 
+def get_model_key(model):
```

Comment:
Hmm, bit unsure if we really need this function? Do we expect many cases when we would call `get_model_key` with something else than `datastore_services.Model`? Because I would say that in a code where I know that I have a `datastore_services.Model` I would use `key` directly.

Reply:
Hmm true I don't think we'll need it after we move to the real datastoreio function, but until then we'd need this function because the only way to query everything is to sort by key (so we use this function to do so).

Reply:
Done: https://github.com/oppia/oppia/pull/12890#discussion_r638281264